addon/components/polaris-drop-zone.js

```diff
@@ -19,18 +20,47 @@ const iconDragDrop = 'drag-drop';
 const iconAlertCircle = 'alert-circle';
 
 export default Component.extend(ContextBoundEventListenersMixin, {
-  attributeBindings: ['ariaDisabled:aria-disabled'],
+  layout,
 
-  classNames: ['Polaris-DropZone'],
+  /**
+   * ID for file input
+   *
+   * @type {String}
+   * @default null
+   * @public
+   * @property id
+   */
+  id: null,
```

Comment:
Not 100% sure how Ember handles passing `id=<whatever>` into a component, but from a faint memory I think it might use that as the `id` of that component's element, which will cause problems if we then try using that same ID on another element. With that in mind, we might be safer renaming this to something like `inputId` (as I think we've done for a bunch of other form-related components) and documenting the difference from the React interface in the component's markdown file.

You could also default this to a CP that returns `guidFor(this)`, which I think we've also done elsewhere, if that cleans up logic later on.

Reply:
I think it should be fine setting the `id` as long as it's unique (we don't re-use it in this component on multiple components - which I don't think we do)
setting a default seems to be done in `getDerivedStateFromProps()`, but agree we have `guidFor` instead of `createUniqueIDFactory`